template/python/main.py

```diff
@@ -1,24 +1,13 @@
 #!/usr/bin/env python
-from http.server import BaseHTTPRequestHandler, HTTPServer
-from handler import handler
-
-class S(BaseHTTPRequestHandler):
-    def _set_headers(self):
-        self.send_response(200)
-        self.end_headers()
 
-    def do_POST(self):
-        self._set_headers()
-        length = int(self.headers['content-length'])
-        data = self.rfile.read(length)
-        self.wfile.write(handler(data))
+from handler import handler
+from flask import Flask, request
+app = Flask(__name__)
 
-def run(server_class=HTTPServer, handler_class=S, port=80):
-    server_address = ('', port)
-    httpd = server_class(server_address, handler_class)
-    print('Starting http server...')
-    httpd.serve_forever()
+@app.route('/', methods=['POST'])
+def main():
+    print(request)
+    data = request.get_data()
+    return handler(data)
```

Comment:
the only danger here is that we can not be sure that the handler will return a type that flask knows how to serialize: see http://flask.pocoo.org/docs/0.12/quickstart/#about-responses Other than that, this looks good . I will probably add a parallel template that supports the classic `stdout` style as well

Reply:
Note, I don't think there is anything to change here